core/templates/dev/head/pages/exploration_player/AudioPreloaderService.js

```diff
@@ -20,117 +20,124 @@ oppia.factory('AudioPreloaderService', [
-      });
-
-      _preloadedLanguageCodes.push(languageCode);
+    var _init = function(exploration) {
+      _exploration = exploration;
     };
 
-    var _showBandwidthConfirmationModal = function(
-        audioTranslationsForContent, languageCode,
-        confirmationCallback) {
-      $modal.open({
-        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
-          '/pages/exploration_player/' +
-          'audio_preload_bandwidth_confirmation_modal_directive.html'),
-        resolve: {},
-        backdrop: true,
-        controller: [
-          '$scope', '$modalInstance',
-          'ExplorationPlayerStateService', 'AudioPreloaderService',
-          'LanguageUtilService',
-          function(
-              $scope, $modalInstance,
-              ExplorationPlayerStateService, AudioPreloaderService,
-              LanguageUtilService) {
-            $scope.fileSizeOfCurrentAudioTranslationMB =
-              audioTranslationsForContent[languageCode]
-                .getFileSizeMB().toPrecision(3);
-            $scope.totalFileSizeOfAllAudioTranslationsMB =
-              ExplorationPlayerStateService.getExploration()
-                .getAllAudioTranslationsFileSizeMB(languageCode)
-                .toPrecision(3);
-            $scope.currentLanguageDescription =
-              LanguageUtilService.getAudioLanguageDescription(languageCode);
-            $scope.shouldDownloadAllAudioInExploration = false;
+    var _getAudioFilesInBfsOrder = function(sourceStateName) {
+      var languageCode = AudioTranslationManagerService
+        .getCurrentAudioLanguageCode();
+      var explorationGraph = ComputeGraphService.compute(
+        _exploration.getInitialState().name, _exploration.getStates());
+      var queue = [];
+      var seen = {};
+      var audioFiles = [];
+      allAudioTranslations = _exploration.getAllAudioTranslations(
+        languageCode);
+      queue.push(sourceStateName);
+      while (queue.length > 0) {
+        var currStateName = queue.shift();
+        var audioFile = allAudioTranslations[currStateName];
+        if (audioFile != null) {
+          audioFiles.push(audioFile.filename);
+        }
+        for (var e = 0; e < explorationGraph.links.length; e++) {
+          var edge = explorationGraph.links[e];
+          var dest = edge.target;
+          if (edge.source === currStateName && !seen.hasOwnProperty(dest)) {
+            seen[dest] = true;
+            queue.push(dest);
+          }
+        }
+      }
+      return audioFiles;
+    };
 
-            $scope.confirm = function() {
-              $modalInstance.close({
-                shouldDownloadAllAudioInExploration:
-                  $scope.shouldDownloadAllAudioInExploration,
-                shouldOpenSettingsModal: false
-              });
-            };
+    var loadAudio = function(audioFile) {
+      AssetsBackendApiService.loadAudio(
+        ExplorationContextService.getExplorationId(), audioFile
```

Comment:
Is audioFile here a file or a filename? If the latter, please rename accordingly.

Ditto for _filesBeingDownloaded -- I'm having a bit of trouble parsing this function because it's not clear to me what the variables represent. It's true that filenamesOfFilesBeingDownloaded is a bit clumsy, though -- maybe filenamesAwaitingDownload or filenamesCurrentlyDownloading might work, but you might have better ideas!

Reply:
Done